pgmax/fg/graph.py

```diff
@@ -807,9 +925,14 @@ def BP(
         \tget_beliefs: Function to calculate beliefs from BPArrays.\n
     """
     max_msg_size = int(np.max(bp_state.fg_state.wiring.edges_num_states))
-    num_val_configs = (
-        int(bp_state.fg_state.wiring.factor_configs_edge_states[-1, 0]) + 1
-    )
+
+    if bp_state.fg_state.wiring.enum_wiring is not None:
```

Comment:
Again, instead of having these kind of if/else, we should probably do a simple loop on a generic structure (e.g. a dictionary, one element for each factor type) inside this function. 

We should make enough abstractions that adding new factor types should not involve updating this function.